app/models/topic.rb

```diff
@@ -830,6 +832,23 @@ def invite_by_email(invited_by, email, group_ids = nil, custom_message = nil)
     Invite.invite_by_email(email, invited_by, self, group_ids, custom_message)
   end
 
+  def invite_existing_muted?(target_user, invited_by)
+    # check if inviter is muted
+    return true if invited_by.id && MutedUser.where(user_id: target_user.id, muted_user_id: invited_by.id)
+        .joins(:muted_user)
```

Comment:
This looks odd to me cause we are joining on the same table. I think you meant to join on `users`?